### DISCUSSION TITLE
Generate 3D coordinates for molecule with non-standard valence

### DISCUSSION THREAD

**Discussion by manu-ha-hqs** [rdkit/rdkit#5907] [Q&A]
I am trying to generate 3D coordinates for molecules that have non-standard valences such as chlorine trifluoride (ClF3).
Obtaining a molecule object from the SMILES string works without sanitization and the property cache can successively be updated as follows:
```
from rdkit import Chem
from rdkit.Chem import AllChem

mol = Chem.MolFromSmiles("F[Cl](F)F", sanitize=False)
mol.UpdatePropertyCache(strict=False)
```
(Omitting `sanitize=False` prints the error message "Explicit valence for atom # 1 Cl, 3, is greater than permitted" and returns `None`, while updating the property cache ensures that `Chem.AddHs(mol)` would not throw an error, which can be important for larger systems.)
However, upon trying to generate 3D coordinates for the molecule, an `AtomValenceException` is thrown with the corresponding error message:
```
>>> status_embed = AllChem.EmbedMolecule(mol)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
rdkit.Chem.rdchem.AtomValenceException: Explicit valence for atom # 1 Cl, 3, is greater than permitted
```
The same error would occur when omitting the `strict=False` option of `UpdatePropertyCache`.
Is there any way to "ignore" this non-standard valence (like setting the sanitize/strict options to False) and obtain 3D coordinates for systems like this one?

**Comment by greglandrum** [comment]
Hi @manu-ha : this is a bug in the way fragments are handled in the embedding code.
Since your molecule doesn't contain disconnected fragments, you can safely turn that off as follows:
```
In [5]: ps = AllChem.ETKDGv3()

In [6]: ps.embedFragmentsSeparately = False

In [7]: AllChem.EmbedMolecule(mol,ps)
Out[7]: 0
```

**Reply by manu-ha-hqs** [reply]
Hi @greglandrum, thank you very much for your quick reply. Your approach works and gives a reasonable result for my example molecule above. However, I noticed some other things I would like to ask about.
Embedding systems like ClF5 or IF7 yields planar structures instead of (bi)pyramidal ones, is there something one can do about that?
Furthermore, for the non-cyclic molecule with SMILES string `"N(=ClCl)Cl"` I got a runtime error:
```
>>> status_embed = AllChem.EmbedMolecule(mol, ps)
RuntimeError: Pre-condition Violation
	RingInfo not initialized
	Violation occurred on line 129 in file Code/GraphMol/RingInfo.cpp
	Failed Expression: df_init
	RDKIT: 2022.03.5
	BOOST: 1_74
```
which I could fix by replacing the `UpdatePropertyCache` call by 
```Chem.SanitizeMol(mol, Chem.SanitizeFlags.SANITIZE_SYMMRINGS)```
Should the call to `Chem.SanitizeMol` in general be preferred over `UpdatePropertyCache`?
